### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by aselbie**

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const scripts    = document.getElementsByTagName('script');
+const src        = scripts[scripts.length - 1].getAttribute('src');
+const publicPath = src.substr(0, src.lastIndexOf('/') + 1);
+
+__webpack_public_path__ = publicPath
```
Comment (vietnogi) [inline app/index.js]: are we using these variables/constants?

File: app/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+require('appirio-tech-ng-login-reg')
+require('appirio-tech-ng-messaging')
+require('appirio-tech-ng-projects')
+require('appirio-tech-ng-status-report')
+require('appirio-tech-ng-submissions')
+require('appirio-tech-ng-submit-work')
+require('appirio-tech-ng-timeline')
+require('appirio-tech-ng-work-layout')
+
+require('./scripts/messaging.controller')
```
Comment (vietnogi) [inline app/index.js]: use reggex to require all controllers?

File: app/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+const scripts    = document.getElementsByTagName('script');
+const src        = scripts[scripts.length - 1].getAttribute('src');
+const publicPath = src.substr(0, src.lastIndexOf('/') + 1);
+
+__webpack_public_path__ = publicPath
+
```
Comment (vietnogi) [inline app/index.js]: can we convert to coffee = )

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// Reference: http://karma-runner.github.io/0.12/config/configuration-file.html
```
Comment (vietnogi) [inline karma.conf.js]: can we share this file among all the other repos?

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+/**
+ * Loaders
+ * Reference: http://webpack.github.io/docs/configuration.html#module-loaders
+ * List: http://webpack.github.io/docs/list-of-loaders.html
+ * This handles most of the magic responsible for converting modules
+ */
+
+// Initialize module
+config.module = {
```
Comment (vietnogi) [inline webpack.config.js]: can we get some of the configs from the web pack config repo?

File: app/index.js (hunk 1 above)
Comment (aselbie) [inline app/index.js]: This one is actually significant to webpack. See https://webpack.github.io/docs/configuration.html#output-publicpath

File: app/index.js (hunk 2 above)
Comment (aselbie) [inline app/index.js]: Good point.

File: app/index.js (hunk 3 above)
Comment (aselbie) [inline app/index.js]: Done.

File: karma.conf.js (hunk 4 above)
Comment (aselbie) [inline karma.conf.js]: Probably. We'll come back to it when we're figuring out the test running part.

File: webpack.config.js (hunk 5 above)
Comment (aselbie) [inline webpack.config.js]: Done.

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+require('./node_modules/coffee-script/register');
```
Comment (vietnogi) [inline webpack.config.js]: don't need this

**Comment by vietnogi** [comment]
lgtm beside the small comment

**Pull request merged**